xtask/src/main.rs

```diff
@@ -63,6 +63,26 @@ impl FromStr for Memory {
     }
 }
 
+#[derive(Clone)]
+enum DramSize {
+    TwoG,
+    FourG,
+    EightG,
+}
+
+impl FromStr for DramSize {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
```

Comment:
```suggestion
        match s.as_str() {
```

Since you are checking upper and lowercase strings below, you can skip the conversion (and save a heap allocation).

Reply:
ah yes - will keep that in mind for subsequent refinement, thank you!